src/Module.Server/Rewards/CrpgRewardServer.cs

```diff
@@ -174,14 +191,27 @@ public override void OnRemoveBehavior()
             if (missionPeer.Team != null && missionPeer.Team.Side != BattleSideEnum.None)
             {
                 bool isValorousPlayer = valorousPlayerIds.Contains(playerId);
-                SetReward(userUpdate, crpgPeer, durationRewarded, isValorousPlayer, defenderMultiplierGain,
-                    attackerMultiplierGain, constantMultiplier);
-                if (updateUserStats)
+                int netCompensationForCrpgUser = netCompensationByCrpgUserId.TryGetValue(crpgUserId, out int compensationForCrpgUser) ? compensationForCrpgUser : 0;
+                if (networkPeer.IsConnectionActive)
                 {
-                    SetStatistics(userUpdate, networkPeer, periodStats);
-                }
+                    SetRewardForConnectedPlayer(userUpdate, crpgPeer, durationRewarded, netCompensationForCrpgUser, isValorousPlayer,
+                        defenderMultiplierGain, attackerMultiplierGain, constantMultiplier);
+
+                    if (updateUserStats)
+                    {
+                        SetStatistics(userUpdate, networkPeer, periodStats);
+                    }
 
-                userUpdate.BrokenItems = BreakItems(crpgPeer, durationRewarded * (lowPopulationServer ? 0.2f : 1f));
+                    userUpdate.BrokenItems = brokenItems[crpgUserId];
+                }
+                else
+                {
+                    userUpdate.Reward = new CrpgUserReward
+                    {
+                        Experience = 0,
+                        Gold = netCompensationForCrpgUser,
+                    };
+                }

```

Comment:
Don't worry, i also do this "professionally" (as in it's my job) i'm used to code review and i fully expected this outcome (or it's not really a suprise). It's my first time modding and first time using C#. Rather i'm glad you give me this opportunity to learn and hope you are not annoyed by me making this many mistakes.

You really think it's okay to "escape" paying compensation if you just disconnect? Probably it's not a big deal, but i really liked the idea of no escape from the compensation system.
I'm not really sure how to handle that situation otherwise, because in `CalculateCompensationByCrpgUserId` i don't really regard if people were spawned or not. Either it's not going to be a sum of 0 at the end or i would need to implement other logic to disable both sides of compensation (compensated & fine).
Your call if it's okay to artificially inflate/deflate the economy (shouldn't be too big but still)

Reply:
What do you think of removing the code handling disconnected players and doing a second PR to handle them?

Reply:
So you saying that asynchronous compensation is okay? Needs to probably be tested to be sure, but right now i log every teamhit and then compensate spawned players anyway.
But yes, i would do it in another PR then